File: engines/polyglot-route-engine/fixtures/enterprise/enterprise_asset_service.py
from dataclasses import dataclass
from typing import Optional
from fastapi import APIRouter, HTTPException

@dataclass
class Asset:
    serial: str
    status: str
    value: float

    def __del__(self):
        pass

router = APIRouter(prefix="/api/v1/assets", tags=["assets"])

@router.get("/{serial}")
async def get_asset_by_serial(serial: str) -> Asset:
    try:
        if not serial:
            raise ValueError("Asset serial is invalid")
        return Asset(serial=serial, status="ACTIVE", value=100.0)
    except Exception as ex:
        raise HTTPException(status_code=500, detail=f"Failed to retrieve asset: {str(ex)}")

@router.post("")
async def create_asset(asset: Asset) -> Asset:
    try:
        return Asset(serial=asset.serial, status=asset.status, value=asset.value)
    except Exception as ex:
        raise HTTPException(status_code=500, detail=f"Failed to create asset: {str(ex)}")

File: engines/polyglot-route-engine/fixtures/enterprise/EnterpriseAssetService.cs
using System;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;

namespace Elmos.Enterprise
{
    public class Asset
    {
        public string Serial { get; set; }
        public string Status { get; set; }
        public double Value { get; set; }

        public Asset(string serial, string status, double value)
        {
            Serial = serial ?? throw new ArgumentNullException(nameof(serial));
            Status = status;
            Value = value;
        }
    }

    [ApiController]
    [Route("api/v1/assets")]
    public class EnterpriseAssetController : ControllerBase
    {
        [HttpGet("{serial}")]
        public async Task<ActionResult<Asset>> GetAssetBySerial(string serial)
        {
            try
            {
                await Task.Yield();
                if (string.IsNullOrWhiteSpace(serial))
                {
                    throw new ArgumentException("Asset serial is invalid");
                }
                return Ok(new Asset(serial, "ACTIVE", 100.0));
            }
            catch (Exception ex)
            {
                return StatusCode(500, $"Failed to retrieve asset: {ex.Message}");
            }
        }

        [HttpPost]
        public async Task<ActionResult<Asset>> CreateAsset([FromBody] Asset asset)
        {
            try
            {
                await Task.Yield();
                return Ok(new Asset(asset.Serial, asset.Status, asset.Value));
            }
            catch (Exception ex)
            {
                return StatusCode(500, $"Failed to create asset: {ex.Message}");
            }
        }
    }
}
